crates/beautifier/src/beautifier.rs

```diff
@@ -0,0 +1,228 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#![deny(
+    dead_code,
+    nonstandard_style,
+    unused_imports,
+    unused_mut,
+    unused_variables,
+    unused_unsafe,
+    unreachable_patterns
+)]
+
+use air_parser::ast;
+
+use std::fmt::Display;
+use std::io;
+
+pub const DEFAULT_INDENT_SIZE: usize = 4;
+
+macro_rules! compound {
+    ($beautifier:expr, $indent:expr, $body:expr) => {{
+        let out = &mut $beautifier.output;
+        $crate::beautifier::fmt_indent(out, $indent)?;
+        writeln!(out, "{}:", $body)?;
+        $crate::beautifier::Beautifier::beautify_walker(
+            $beautifier,
+            &$body.instruction,
+            $indent + $beautifier.indent_size,
+        )
+    }};
+}
+
+macro_rules! multiline {
+    ($beautifier:expr, $indent:expr $(; $fmt1:literal $(, $arg1:expr)*; $nest:expr)+) => ({
+        let step = $beautifier.indent_size;
+        $({
+              let out = &mut $beautifier.output;
+              $crate::beautifier::fmt_indent(out, $indent)?;
+              writeln!(out, $fmt1 $(, $arg1)*)?;
+          }
+          $crate::beautifier::Beautifier::beautify_walker($beautifier, $nest, $indent + step)?;
+        )+
+        Ok(())
+    });
+}
+
+fn fmt_indent(output: &mut impl io::Write, indent: usize) -> io::Result<()> {
+    write!(output, "{:indent$}", "", indent = indent)
+}
+
+struct CallArgs<'ctx, 'i>(&'ctx [ast::Value<'i>]);
+
+impl<'ctx, 'i> Display for CallArgs<'ctx, 'i> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        use itertools::Itertools as _;
+
+        f.write_fmt(format_args!("{}", self.0.iter().format(", ")))
+    }
+}
+
+struct CallTriplet<'ctx, 'i>(&'ctx ast::Triplet<'i>);
+
+impl<'ctx, 'i> Display for CallTriplet<'ctx, 'i> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_fmt(format_args!(
+            "{} ({}, {})",
+            self.0.peer_pk, self.0.service_id, self.0.function_name
+        ))
+    }
+}
+
+/// Error produced by the Beautifier.
+#[derive(Debug, thiserror::Error)]
+pub enum BeautifyError {
+    #[error("{0}")]
+    Parse(String),
+    #[error(transparent)]
+    Io(#[from] io::Error),
+}
+
+/// AIR beautifier.
+pub struct Beautifier<W: io::Write> {
+    output: W,
+    indent_size: usize,
+}
+
+impl<W: io::Write> Beautifier<W> {
+    /// Beautifier for the output with default indent size.
+    pub fn new(output: W) -> Self {
+        Self {
+            output,
+            indent_size: DEFAULT_INDENT_SIZE,
+        }
+    }
+
+    /// Beautifier for the output with custom indent size.
+    pub fn new_with_indent(output: W, indent_step: usize) -> Self {
```

Comment:
We have several similar terms in this file:
`indent_step`,
`step`,
`indent`,
`indent_size`,

Can we reduce their count somehow?